### PR TITLE
Add documentation for UDPSocket class

### PR DESCRIPTION
**Pull request by davydovanton**

### CODE REVIEW

File: src/socket/udp_socket.cr [hunk 1]
```diff
@@ -1,5 +1,39 @@
...
+# This simple protocol provides transport layer addressing in the form of UDP ports and an
+# optional checksum capability.
+#
+# UDP is a very simple protocol. Messages, so called datagrams, are sent to other hosts on
+# an IP network without the need to set up special transmission channels or data paths
+# beforehand. The UDP socket only needs to be opened for communication. It listens for
+# incoming messages and sends outgoing messages on request.
+#
+# This implementation supports both IPv4 and IPv6 addresses. For IPv4 adresses you need use
+# `Socket::Family::INET` famyly (used by default). And `Socket::Family::INET6` for IPv6
```
Comment (ysbaddaden) [inline src/socket/udp_socket.cr]: Small typo: "famyly"

**Comment by vyp** [comment]
Typos: "famyly" should be "family", and "connects" in "Attempts to connects the socket to a remote address" near the bottom of the diff should be "connect".

File: src/socket/udp_socket.cr (hunk 1 above)
Comment (davydovanton) [inline src/socket/udp_socket.cr]: nice catch, thanks!

**Comment by davydovanton** [comment]
@vyp you're right, thanks for catch :+1:

**Comment by ysbaddaden** [comment]
:+1:

**Comment by asterite** [comment]
Awesome, thanks!!

**Pull request merged**